plugin/path_creds.go

```diff
@@ -141,10 +141,7 @@ func (b *backend) credReadOperation(ctx context.Context, req *logical.Request, f
 		}
 	}
 	if respErr != nil {
-		return nil, err
-	}
-	if resp == nil {
```

Comment:
Should a nil response result in an error?

Reply:
I checked through all the ways the resp is populated, and found that it's impossible for resp to be nil.

It's either populated directly with a resp, or via generateAndReturnCreds, which never returns a nil resp. So we should be good.